zerver/tornado/event_queue.py

```diff
@@ -251,6 +255,15 @@ def accepts_event(self, event: Mapping[str, Any]) -> bool:
             # events are sent only if user_settings_object is False,
             # otherwise only 'user_settings' event is sent.
             return False
+
+        if (
+            event["type"] == "realm_user"
+            and event["op"] == "add"
+            and self.supports_incomplete_realm_users
+            and event["person"]["full_name"] == str(UserProfile.INACCESSIBLE_USER_NAME)
```

Comment:
Yeah this is not ideal; it might even just fail to work on account of i18n issues. One option is to put an extra field in the event that gets popped in a function along the lines of `process_custom_profile_fields_event` and used to know that this is an event that should only be put into Tornado queues that lack the client capability.

```
diff --git a/zerver/actions/create_user.py b/zerver/actions/create_user.py
index 51c4f54898..8f4b3a732f 100644
--- a/zerver/actions/create_user.py
+++ b/zerver/actions/create_user.py
@@ -394,6 +394,7 @@ def notify_created_user(user_profile: UserProfile, notify_user_ids: List[int]) -
             type="realm_user",
             op="add",
             person=get_data_for_inaccessible_user(user_profile.realm, user_profile.id),
+            inaccessible_user=True,
         )
         send_event_on_commit(user_profile.realm, event, user_ids_without_access_to_created_user)
 
```